### ISSUE TITLE
Design System: `DataList` dialog does not have `aria-label`

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#10919]
## Bug Description

I was getting some aXe errors on the font picker due to missing `aria-label` / `aria-labelledby` / `title` on the `<Container>` component.

Docs on `role="dialog"`: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

https://github.com/GoogleForCreators/web-stories-wp/blob/fe78d62bf89ef320638c1f66984a91c1291e521c/packages/design-system/src/components/datalist/container.js#L117-L123

## Expected Behaviour

`DataList` dialog should not cause aXe errors due to missing label / title.

It should be possible to pass a label for that container, e.g. "Font picker"

## Steps to Reproduce

1. Open font picker
2. Run aXe
3. See error

## Additional Context

- Plugin Version: 1.18.0
- WordPress Version: 5.9
- Operating System: macOS
- Browser: Chrome

**Comment by felipebochehin87** [comment]
Tested against https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin, using PR #11013.

Regressed dropdowns (font, publisher logo, author, visibility, publish date) - working as expected.
